model/schema.json

```diff
@@ -647,99 +645,152 @@
+                  },
+                  "status": {
+                    "description": "The current status of the input",
+                    "type": "string",
+                    "enum": ["healthy", "error", "degraded"]
+                  },
+                  "message": {
+                    "description": "The current status message of the intput",
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "id",
+                  "template_id",
+                  "status",
+                  "message"
+                ]
+              }
+            }
+          },
+          "required": [
+            "id",
+            "revision",
+            "inputs"
+          ]
+        }
       },
-      "policy": {
-        "title": "Checkin Policy",
-        "description": "The current status of the applied policy",
+      "required": [
+        "_id",
+        "agent",
+        "status",
+        "message",
+        "enrolled_at",
+        "updated_at"
+      ]
+    },
+
+    "policy-data": {
+        "title": "Policy Data",
+        "description": "The policy data that an agent needs to run",
         "type": "object",
         "properties": {
-          "id": {
-            "description": "The ID for the policy",
-            "type": "string",
-            "format": "uuid"
-          },
-          "revision": {
-            "description": "The revision of the policy",
-            "type": "integer"
-          },
-          "inputs": {
-            "title": "Checkin Policy Input",
-            "description": "The current input status per policy",
-            "type": "array",
-            "items": {
-              "type": "object",
-              "properties": {
-                "id": {
-                  "description": "The ID for the input",
-                  "type": "string",
-                  "format": "uuid"
-                },
-                "template_id": {
-                  "description": "The template ID for the input",
-                  "type": "string",
-                  "format": "uuid"
-                },
-                "status": {
-                  "description": "The current status of the input",
-                  "type": "string",
-                  "enum": ["healthy", "error", "degraded"]
-                },
-                "message": {
-                  "description": "The current status message of the intput",
-                  "type": "string"
-                }
-              },
-              "required": [
-                "id",
-                "template_id",
-                "status",
-                "message"
-              ]
+            "id": {
+                "description": "The policy's ID",
+                "type": "string"
+            },
+            "outputs": {
+                "description": "A map of all outputs that the agent running the policy can use to send data to.",
+                "format": "raw"
```

Comment:
@juliaElastic, If we define additional keys as part of the policy definition it should remove some of the boilerplate around casting for the remote output definition.
But we wound need to decide on how much detail we want to get into (should fleet-server care or be aware of attributes it does not use? should we use the `additionalProperties` to hide some of these?)

Reply:
I did a little expirementing and was unable to get the json schema generator tool we use to output generic objects (type `map[string]interface`), but it's easily able to do lists of type `[]interface{}`, or just assign it the generic `interface{}` type.

That means if we want to have a stricter definition in fleet-server then a `json.RawMessage` or `interface{}` for outputs we would need to define the properties that exist in the policy in Elasticsearch (the schema we return from our openapi endpoint can differ by removing the secret_token for instance)

Reply:
Outputs is tracked as a `map[string]map[string]interface{}` to make it easy to iterate in the fleet-server.
I didn't define attributes in outputs as it can get complex (ssl, proxies, keys, etc may be involved)